frontend/src/features/videos/VideoApi.tsx

```diff
@@ -1,12 +1,11 @@
 import { useState, useEffect } from 'react';
 
-import { VideoSerializerWithCriteria as Video } from 'src/services/openapi/models/VideoSerializerWithCriteria';
+import {
+  VideoService,
+  VideoSerializerWithCriteria as Video,
+} from 'src/services/openapi';
 
-const api_url = process.env.REACT_APP_API_URL;
-const client_id = process.env.REACT_APP_OAUTH_CLIENT_ID || '';
-const client_secret = process.env.REACT_APP_OAUTH_CLIENT_SECRET || '';
-
-const _inMemoryCache: { [videoId: string]: Video } = {};
+const _inMemoryCache: { [videoId: string]: Promise<Video> } = {};
```

Comment:
Not directly related to this PR, but nice to have: the cache values are now  `Promise<Video>`. As soon as a request is executed, the promise is present in cache, and the multiple requests to the same video are no longer duplicated.

Reply:
Thanks! Looks good